app/controllers/forgot_passwords_controller.rb

```diff
@@ -0,0 +1,42 @@
+class ForgotPasswordsController < ApplicationController  
+  skip_before_action :require_user
+
+  def create
+    user = User.find_by(email: params[:email])
+    if user
+      user.generate_token!
+      token = user.reload.token
```

Comment:
Oops, `update_attribute`, not `update`. I would still think it affects the in-memory object though. I'm curious if it would work without the `reload`.